packages/wdio-browserstack-service/src/launcher.ts

```diff
@@ -195,11 +196,14 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
             PercyLogger.error(`Error while setting best platform for Percy snapshot at worker start ${err}`)
         }
 
-        if (!isBrowserstackInfra(this._options)) {
+        if (!isBrowserstackInfra(this._config) && SUPPORTED_BROWSERS_FOR_AI.includes(caps.browserName) ) {
```

Comment:
1. Possible to move this logic to healing helper function?
2. Does it need considerations for NL2Steps - where that logic needs to be triggered for browserstack infra / non browserstack infra both ?